core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -169,7 +169,7 @@ object ZkUtils {
   }
 
   def getReassignmentJson(partitionsToBeReassigned: Map[TopicPartition, Seq[Int]]): String = {
-    Json.encodeAsString(Map(
+    Json.legacyEncodeAsString(Map(
```

Comment:
Why is this needed?

Reply:
when we use Json.encodeAsString,  Scala Map is getting encoded as ```{"empty":false,"traversableAgain":true}```

Related link: https://stackoverflow.com/questions/27948128/how-to-convert-scala-map-into-json-string

Reply:
We should pass a Java map.